external/fv3fit/fv3fit/train.py

```diff
@@ -156,6 +162,7 @@ def main(args, unknown_args=None):
         args.validation_data_config,
         args.local_download_path,
         variable_names=training_config.variables,
+        in_memory=training_data_config.kwargs.get("in_memory", False),
```

Comment:
I think you might need to move this to a top-level configuration option. Whether the BatchesLoader should use in-memory is a separate question from whether the cached batches_from_netcdf data should be kept in-memory. When the full dataset doesn't fit in memory, we'd need the batches config to have `in_memory=False` even if we want the cached (probably subsampled) data files to load with `in_memory=True`.

There are a couple ways you can present this in the config. You could group local_download_path and in_memory into their own `CacheConfig` dataclass. If you keep them as separate flat options, I'd suggest renaming `in_memory` to something that mentions the cache, so it's different from the BatchesLoader kwarg option.

Reply:
Good point. I removed the command line arg `--local-download-path` and there is now a `CacheConfig` class with attributes `in_memory` and `local_download_path`.